src/build/__main__.py

```diff
@@ -48,10 +48,13 @@ def _format_dep_chain(dep_chain):  # type: (Sequence[str]) -> str
 
 def _build_in_isolated_env(builder, outdir, distributions):
     # type: (ProjectBuilder, str, List[str]) -> None
-    with IsolatedEnvBuilder() as env:
-        builder.python_executable = env.executable
-        env.install(builder.build_dependencies)
-        for distribution in distributions:
+    for distribution in distributions:
+        with IsolatedEnvBuilder() as env:
+            builder.python_executable = env.executable
+            # first install the build dependencies
+            env.install(builder.build_dependencies)
+            # then get the extra required dependencies from the backend
+            env.install(builder.get_dependencies(distribution))
```

Comment:
What will happen here if the backend returns some non-PEP-440 compliant specs?

Reply:
PEP 517 says

> This hook MUST return an additional list of strings containing PEP 508 dependency specifications

PEP 508 says

> Versions may be specified according to the PEP-440 [4] rules

Reply:
https://www.python.org/dev/peps/pep-0508/#grammar

Reply:
I know what they must, but if they mess it up will we handle it graciously, and by graciously I mean error out and show the adequate error?

Reply:
Right now we pass it down to pip and let it handle it. So even with #192 the user would see something like.

```
ERROR: Could not find a version that satisfies the requirement ...
ERROR: No matching distribution found for ...
```

I am not opposed to use pypa/packaging (only in the isolation build!) to should a custom error, but I do not think it should block this.

Reply:
As long as pip displays an adequate error I'm fine, but we can work on making clearer the source of the error. Because in this case, the error will be in the backend configuration files... Though doesn't have to be in this PR.